packages/expect/src/matchers.ts

```diff
@@ -391,48 +392,65 @@ const matchers: MatchersObject = {
       );
     }
 
-    const isArray = Array.isArray(received); // has indexOf
-    const isString = typeof received === 'string'; // has indexOf
-    const converted: any =
-      isArray || isString ? received : Array.from(received);
+    if (typeof received === 'string') {
+      const index = received.indexOf(String(expected));
```

Comment:
too lazy to check, but this reads to be like

`expect('hello null in the middle').toContain(null)` would pass? Since `String(null) === 'null'`

Reply:
Yes, good eyes. It is for correct typing of dubious behavior.

1. Do you think it should throw matcher error if received is string and expected is not?
2. If yes, is that a breaking change that we wait to add at Jest 25.

Reply:
it fails today, doesn't it? I'd say it's a bug. If an expect fails with "matcher error" or "null is not in string 'null'" shouldn't matter?

Or does `expect('hello null in the middle').toContain(null)` pass today?

Reply:
I will double-check but the original code is `.indexOf(value)` which implicitly coerces to string:

```js
  test('null', () => {
    expect('isnullornot'.indexOf(null)).toBe(2);
  });
```

Reply:
Oh, I didn't know `indexOf` coerced to string! TIL. Same for `includes` actually (`'isnullornot'.includes(null) ===  true`). I'd still say it's a false positive, but I don't feel strongly about it

Reply:
I agree that it seems unintended loop hole, so you think go ahead and throw matcher error?

Reply:
_I_ am. It might be best to put it in a major, though?

/cc @jeysal @thymikee

Reply:
+1 for matcher error, but also think it's breaking and belongs in a major

Reply:
What do you think if I include the code and tests commented out, with a comment that we can find: `// BREAKING change to include in Jest 25`

Reply:
For tracking the breaking change to be included, we should create an issue assigned to the milestone like https://github.com/facebook/jest/issues/7749.
I'd prefer not to put significant amounts of commented out code into master, if you've already written it maybe push it to a branch and link that in the issue so it doesn't get lost?

Reply:
feel free to open one - we can stick all the tiny breakages we wanna make there

Reply:
Created https://github.com/facebook/jest/issues/8023

Reply:
@SimenB Do you mean one for all of them? I think I'd prefer separate issues unless there is a group of breaking changes that have something in common. The small issues will already be grouped using the milestone.